spec/apis/path-to-active-when.spec.js

```diff
@@ -2,182 +2,121 @@ import { pathToActiveWhen } from "single-spa";
+      "http://app.com/pathname/": true,
+      "http://app.com/pathname/?query-string=1": true,
+      "http://app.com/pathname/extra": true,
+      "http://app.com/pathname": false,
+      "http://app.com/pathname?query-string=1": false,
+    });
+
+    expectPathToMatch("/pathname/:dynamic/", {
+      "http://app.com/pathname/123/": true,
+      "http://app.com/pathname/123/?query-string=1": true,
+      "http://app.com/pathname/123/extra": true,
+      "http://app.com/pathname/123/extra?query-string=1": true,
+      "http://app.com/pathname/123": false,
+      "http://app.com/pathname/123?query-string=1": false,
+    });
+
+    expectPathToMatch("/#/pathname", {
+      "http://app.com/#/pathname": true,
+      "http://app.com/#/pathname?query-string=1": true,
+      "http://app.com/#/pathname/": true,
+      "http://app.com/#/pathname/?query-string=1": true,
+      "http://app.com/#/pathname/anything/everything": true,
+      "http://app.com/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1#/pathname/anything/everything?query-string=1": true,
+    });
+
+    expectPathToMatch("/#/pathname/:dynamic/notDynamic", {
+      "http://app.com/#/pathname/1/notDynamic": true,
+      "http://app.com/#/pathname/1/notDynamic/": true,
+      "http://app.com/#/pathname/1/notDynamicExtra": false,
+      "http://app.com/#/pathname//notDynamic/anything/everything": false,
+    });
+
+    expectPathToMatch("/pathname/:dynamic/notDynamic", {
+      "http://app.com/pathname/1/notDynamic": true,
+      "http://app.com/pathname/1/notDynamic/anything/everything": true,
+      "http://app.com/pathname//notDynamic": false,
+    });
+
+    expectPathToMatch("", {
+      "http://app.com": true,
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/", {
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic", {
+      "http://app.com/1/1": true,
+      "http://app.com/1/1/": true,
+      "http://app.com/1//": false,
+      "http://app.com/1/": false,
+      "http://app.com/1": false,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic/anything/everything", {
+      "http://app.com/1/1/anything/everything": true,
+    });
+
+    expectPathToMatch("/pathname#/subpath/:dynamic", {
+      "http://app.com/pathname#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1/with/other/things": false,
+    });
+
+    expectPathToMatch("/#/subpath/:dynamic", {
+      "http://app.com/#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1": true,
+      "http://app.com/subpath/1/with/other/things": false,
+    });
+
+    // Impossible paths due to URL always goin from root 'app.com' -> 'app.com/'
+    expectPathToMatch("#/subpath/:dynamic", {
```

Comment:
As far as I searched, we the location will always have the `/` prefix, so when we declare an activeWhen with `#/subpath` the URL will automatically becomes `app.com/#/subpath` (only if it's root). I'm not sure what to do, I'm really open to some help here. Is it possible to the following asserts be true?

Reply:
This?

```js
if (activeWhen.startsWith('#')) {
  activeWhen = '/' + activeWhen
}
```

Reply:
Yeah, I'm not sure how to handle those cases, was waiting for some feedback on what to do!

When testing just the regex generator in isolation '#/subpath/:dynamic' really generates a regex able to match strings like '#/subpath/1
But when testing the actual pathToActiveWhen (that needs a URL as argument) this test would never pass (even without my changes) making me wonder if paths without starting with `/` ever worked in real scenarios (based on how `location` object works):

Just to be extra clear, I'm thinking about this scenario:
I declared an App with activeWhen: `'#/app1'`
In the browser, when I go to `https://bleus.com/#/app1`, the App would never mount (because the regex generated by `#/app1` wouldn't match `https://bleus.com/#/app1`)

When developing, `new Url('https://bleus.com#/app1')` would give me a location.href == 'https://bleus.com/#/app1', automatically adding a slash between `com` and `#`. So when we do:
``` js
location.href
      .replace(location.origin, "")
```
it would never (as far as I know at least) give me `#/app1` but always somethings starting with `/`...

Reply:
I think it might be best to automatically prepend a `/` when we see an `activeWhen` that starts with `#`. That way our users don't have to worry about this nuanced edge case.